src/pybind/mgr/dashboard/services/nvmeof_conf.py

```diff
@@ -58,3 +59,33 @@ def remove_gateway(cls, name):
             raise NvmeofGatewayDoesNotExist(name)
         del config['gateways'][name]
         cls._save_config(config)
+
+    @classmethod
+    def get_service_info(cls):
+        try:
+            config = cls.get_gateways_config()
+            service_name =  list(config['gateways'].keys())[0]
+            addr = config['gateways'][service_name]['service_url']
+            return service_name, addr
+        except KeyError as e:
+            raise DashboardException(
+                msg=f'NVMe-oF configuration is not set: {e}',
+            )
```

Comment:
If no item `[0]` this will through an `IndexError`. Is that a possibility? You could alternatively use `next(list(...), default)` or just capture the `IndexError` exception too.

Reply:
went with capturing the IndexError exception